src/python/pants/backend/docker/target_types.py

```diff
@@ -429,7 +430,7 @@ class DockerImageTarget(Target):
     )
 
 
-@union
+@union(in_scope_types=[EnvironmentName])
```

Comment:
> I imagine we need it because there are no restrictions on what plugin implementations may do inside - like they may run a Process etc?

Yea, that's correct. Given that we're going to consume the environment to run a `Process`, you wouldn't be able to run a `Process` without this marker.

> How do we reason about when to set this and when not to?

As it stands, everything that is marked is marked either because:
1. I guessed that it might need to download a tool (the only place we use the `Platform` currently)
2. I needed to fix a `@rule` graph error that indicated (obtusely) that the `EnvironmentName` was not in scope.